bin/weewx/drivers/vantage.py

```diff
@@ -344,33 +353,62 @@ def queued_bytes(self):
             self.socket.settimeout(self.timeout)
         return length
 
-    def read(self, chars=1):
+    def read(self, chars=1, flush=False):
         """Read bytes from WeatherLinkIP"""
         import socket
         _buffer = ''
+        _recv = ''
         _remaining = chars
         while _remaining:
             _N = min(4096, _remaining)
             try:
                 _recv = self.socket.recv(_N)
-            except (socket.timeout, socket.error), ex:
+            except socket.timeout:
```

Comment:
I did not have a socket error before on my site, so I don't know if it has any use to 'cure' this with a reinvoke of the DMPAFT or LOOP routines. My first idea was: it doesn't solve anything. Just pure theory, though.
The time outs (with 0 bytes received) occur frequently: about one per 100 read records (= one per 25 read actions).
Instead the 'magic numbers' we would better add parameters 'allow timeout' to the read routines, defaulting to False.